utils/utils.go

```diff
@@ -33,11 +33,23 @@ func ParseArgs() cliArgs {
 	}
 }
 
-func RemoveIndexFromSlice(slice []string, idx int) ([]string, error) {
-
-	if idx < 0 || idx >= len(slice) {
-		return slice, errors.New("index out of range")
+func Complement[T comparable](slice, toRemove []T) []T {
```

Comment:
```suggestion
func Complement[T comparable](slice []T, toRemove []T) []T {
```
Personal preference, feel free to ignore.